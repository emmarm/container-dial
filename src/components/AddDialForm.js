import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as actions from '../actions';
import getFavicon from '../utils/getFavicon';

class AddDialForm extends Component {
  state = {
    siteName: '',
    siteUrl: ''
  };

  onSiteNameChange = e => {
    const siteName = e.target.value;
    this.setState(() => ({ siteName }));
  };

  onSiteUrlChange = e => {
    const siteUrl = e.target.value;
    this.setState(() => ({ siteUrl }));
  };

  handleSubmit = async e => {
    e.preventDefault();
    const { siteName, siteUrl } = this.state;
    const { container, addDial, handleHideAddDialModal } = this.props;
    const favicon = await getFavicon(siteUrl);
    const dial = {
      siteName,
      siteUrl,
      container: container.name,
      favicon
    };

    addDial(dial);
    handleHideAddDialModal(e);
  };

  render() {
    return (
      <form className="add-dial-modal__form">
        <div className="add-dial-modal__field">
          <label htmlFor="site-name" className="field__label">
            Site Name
          </label>
          <input
            id="site-name"
            className="field__input"
            type="text"
            placeholder="e.g. Facebook"
            onChange={this.onSiteNameChange}
            value={this.state.siteName}
          />
        </div>

        <div className="add-dial-modal__field">
          <label htmlFor="site-url" className="field__label">
            Site URL
          </label>
          <input
            id="site-url"
            className="field__input"
            type="text"
            placeholder="e.g. https://facebook.com"
            onChange={this.onSiteUrlChange}
            value={this.state.siteUrl}
          />
        </div>

        <div className="add-dial-modal__buttons">
          <button
            onClick={this.props.handleHideAddDialModal}
            className="add-dial-modal__button button--secondary"
          >
            Cancel
          </button>

          <button
            className="add-dial-modal__button button--primary"
            onClick={this.handleSubmit}
            style={{ backgroundColor: this.props.theme.primary }}
          >
            Add Dial
          </button>
        </div>
      </form>
    );
  }
}

AddDialForm.propTypes = {
  handleHideAddDialModal: PropTypes.func.isRequired,
  theme: PropTypes.objectOf(PropTypes.string)
};

const mapStateToProps = state => ({
  container: state.container
});

export default connect(
  mapStateToProps,
  actions
)(AddDialForm);
