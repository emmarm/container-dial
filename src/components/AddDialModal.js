import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';

const AddDialModal = ({ handleHideAddDialModal, container, isOpen, theme }) => (
  <Modal
    isOpen={isOpen}
    onRequestClose={handleHideAddDialModal}
    closeTimeoutMS={150}
    className="add-dial-modal"
  >
    <h2 className="add-dial-modal__title">
      New{' '}
      <span className="title__span" style={{ color: theme.primary }}>
        {container.name}
      </span>{' '}
      Dial
    </h2>

    <form className="add-dial-modal__form">
      <div className="add-dial-modal__field">
        <label htmlFor="site-name" className="field__label">
          Site Name
        </label>
        <input
          id="site-name"
          className="field__input"
          type="text"
          placeholder="e.g. Facebook"
        />
      </div>

      <div className="add-dial-modal__field">
        <label htmlFor="site-url" className="field__label">
          Site URL
        </label>
        <input
          id="site-url"
          className="field__input"
          type="text"
          placeholder="e.g. https://facebook.com"
        />
      </div>

      <div className="add-dial-modal__buttons">
        <button
          onClick={handleHideAddDialModal}
          className="add-dial-modal__button button--secondary"
        >
          Cancel
        </button>

        <button
          className="add-dial-modal__button button--primary"
          style={{ backgroundColor: theme.primary }}
        >
          Add Dial
        </button>
      </div>
    </form>
  </Modal>
);

AddDialModal.propTypes = {
  handleHideAddDialModal: PropTypes.func.isRequired,
  isOpen: PropTypes.bool.isRequired,
  container: PropTypes.objectOf(PropTypes.string)
};

export default AddDialModal;
