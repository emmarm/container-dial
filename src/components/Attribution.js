import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

const Text = styled('p')({
  color: 'white',
  fontSize: 14,
  fontStyle: 'italic',
  fontWeight: 100,
  marginRight: 40,
  textAlign: 'right'
});

const Link = styled('a')(
  {
    textDecoration: 'none'
  },
  ({ theme }) => ({
    color: theme.dark
  })
);

const Attribution = ({ image }) => {
  const { user } = !!image && image;
  return (
    <Text>
      Photo by{' '}
      <Link
        href={`${user &&
          user.links.html}?utm_source=container_dial&utm_medium=referral`}
      >
        {user && user.first_name} {user && user.last_name}
      </Link>{' '}
      on{' '}
      <Link href="https://unsplash.com/?utm_source=container_dial&utm_medium=referral">
        Unsplash
      </Link>
    </Text>
  );
};

Attribution.propTypes = {
  image: PropTypes.object
};

export default Attribution;
