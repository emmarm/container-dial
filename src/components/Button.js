import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

import Icon from './Icon';

const Btn = styled('button')(
  {
    alignItems: 'center',
    border: 'none',
    borderRadius: 4,
    boxShadow:
      'rgba(50, 50, 93, 0.1) 0px 7px 14px, rgba(0, 0, 0, 0.08) 0px 3px 6px',
    color: 'white',
    cursor: 'pointer',
    display: 'flex',
    fontSize: 20,
    fontWeight: 100,
    height: 50,
    justifyContent: 'center',
    padding: '0 25px',
    ':disabled': {
      backgroundColor: '#777',
      cursor: 'auto'
    }
  },
  ({ danger, narrow, padded, primary, theme }) => ({
    backgroundColor: primary ? theme.primary : danger ? '#FF3762' : '#bbb',
    padding: narrow && '0 15px',
    paddingLeft: padded && 30,
    ':hover': {
      backgroundColor: primary ? theme.light : danger ? '#FF5F81' : '#c8c8c8',
      ':disabled': {
        backgroundColor: '#777'
      }
    }
  })
);

const Button = ({
  className,
  danger,
  disabled,
  icon,
  narrow,
  onClick,
  primary,
  submit,
  text
}) => (
  <Btn
    className={className}
    danger={danger}
    disabled={disabled}
    onClick={onClick}
    narrow={narrow}
    padded={!!icon && !!text}
    primary={primary}
    type={submit ? 'submit' : 'button'}
  >
    {text}
    {icon && <Icon icon={icon} padded={!!text} />}
  </Btn>
);

Button.propTypes = {
  className: PropTypes.string,
  danger: PropTypes.bool,
  disabled: PropTypes.bool,
  icon: PropTypes.string,
  onClick: PropTypes.func,
  primary: PropTypes.bool,
  submit: PropTypes.bool,
  text: PropTypes.string
};

export default Button;
