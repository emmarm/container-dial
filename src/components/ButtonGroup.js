import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

const Group = styled('div')({
  display: 'flex',
  justifyContent: 'space-around',
  margin: '25px'
});

const ButtonGroup = ({ children }) => <Group>{children}</Group>;

ButtonGroup.propTypes = {
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.object])
};

export default ButtonGroup;
