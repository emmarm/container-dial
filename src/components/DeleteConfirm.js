import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as actions from '../actions';

export const DeleteConfirm = ({
  dial,
  toggleShowDeleteConfirm,
  handleDelete,
  container,
  theme
}) => (
  <div>
    <p>
      Delete {dial.siteName} from {container.name}?
    </p>

    <button onClick={toggleShowDeleteConfirm}>Cancel</button>

    <button onClick={handleDelete}>Delete</button>
  </div>
);

DeleteConfirm.propTypes = {
  dial: PropTypes.objectOf(PropTypes.string),
  toggleShowDeleteConfirm: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
  container: PropTypes.objectOf(PropTypes.string),
  theme: PropTypes.objectOf(PropTypes.string)
};

const mapStateToProps = state => ({
  container: state.container,
  theme: state.theme
});

export default connect(
  mapStateToProps,
  actions
)(DeleteConfirm);
