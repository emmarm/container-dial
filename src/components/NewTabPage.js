import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { ThemeProvider } from 'emotion-theming';

import * as actions from '../actions';
import Page from './Page';
import GridList from './GridList';
import DialModal from './DialModal';

const DEFAULT_STATE = { showDialModal: false };

export class NewTabPage extends React.Component {
  state = DEFAULT_STATE;

  componentDidMount() {
    const { container, setContainer, startSetBackground } = this.props;

    setContainer(container);
    startSetBackground(container);
  }

  handleShowDialModal = dial => {
    dial.siteName && this.setState(() => ({ dial }));

    this.setState(() => ({
      showDialModal: true
    }));
  };

  handleHideDialModal = e => {
    e && e.preventDefault();
    this.props.setCurrentDial(null);
    this.setState(() => DEFAULT_STATE);
  };

  render() {
    const { container, theme } = this.props;
    return (
      <ThemeProvider theme={theme}>
        <Page>
          <DialModal
            handleHideDialModal={this.handleHideDialModal}
            container={container}
            isOpen={this.state.showDialModal}
          />
          <GridList handleShowDialModal={this.handleShowDialModal} />
        </Page>
      </ThemeProvider>
    );
  }
}

NewTabPage.propTypes = {
  container: PropTypes.objectOf(PropTypes.string),
  theme: PropTypes.objectOf(PropTypes.string),
  dials: PropTypes.arrayOf(PropTypes.object),
  setContainer: PropTypes.func.isRequired,
  setCurrentDial: PropTypes.func.isRequired,
  startSetBackground: PropTypes.func.isRequired
};

export default connect(
  undefined,
  actions
)(NewTabPage);
