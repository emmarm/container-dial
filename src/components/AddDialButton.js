import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

import Button from './Button';

const AddButton = styled(Button)({
  background: 'rgba(255, 255, 255, 0.3)',
  borderRadius: 0,
  height: '100%',
  ':hover': {
    background: 'rgba(255, 255, 255, 0.5)'
  }
});

const AddDialButton = ({ handleShowDialModal }) => (
  <AddButton onClick={handleShowDialModal} text="Add New" icon="add" />
);

AddDialButton.propTypes = {
  handleShowDialModal: PropTypes.func.isRequired
};

export default AddDialButton;
