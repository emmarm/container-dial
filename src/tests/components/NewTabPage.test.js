import React from 'react';
import { shallow } from 'enzyme';

import { NewTabPage } from '../../components/NewTabPage';

describe('NewTabPage component', () => {
  const props = {
    container: { cookieStoreId: '123' },
    theme: { primary: '#333', dark: '#000', light: '#fff' },
    setContainer: jest.fn(),
    setCurrentDial: jest.fn(),
    setTheme: jest.fn(),
    startSetBackground: jest.fn()
  };
  const wrapper = shallow(<NewTabPage {...props} />);

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls setContainer and startSetBackground functions', () => {
    const { setContainer, startSetBackground } = props;
    expect(setContainer).toHaveBeenCalled();
    expect(startSetBackground).toHaveBeenCalled();
  });

  it('does not show DialModal when showDialModal false', () => {
    const dialModal = wrapper.find('Connect(DialModal)');
    expect(dialModal.props().isOpen).toBeFalsy();
  });

  it('shows DialModal when showDialModal true', () => {
    wrapper.setState({ showDialModal: true });
    const dialModal = wrapper.find('Connect(DialModal)');
    expect(dialModal.props().isOpen).toBeTruthy();
  });
});
