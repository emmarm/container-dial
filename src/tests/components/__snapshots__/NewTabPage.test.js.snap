// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTabPage component matches snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "dark": "#000",
      "light": "#fff",
      "primary": "#333",
    }
  }
>
  <Connect(Component)>
    <Connect(DialModal)
      container={
        Object {
          "cookieStoreId": "123",
        }
      }
      handleDeleteDial={[Function]}
      handleHideDialModal={[Function]}
      isOpen={false}
    />
    <Connect(Component)
      dials={
        Array [
          Object {
            "container": "123",
            "favicon": "https://www.udemy.com/staticx/udemy/images/v6/favicon-196x196.png",
            "id": 1,
            "siteName": "Udemy",
            "siteUrl": "www.udemy.com",
            "sortIndex": 0,
          },
          Object {
            "container": "123",
            "favicon": "https://s.ytimg.com/yts/img/favicon_96-vflW9Ec0w.png",
            "id": 2,
            "siteName": "Youtube",
            "siteUrl": "www.youtube.com",
            "sortIndex": 1,
          },
          Object {
            "container": "123",
            "favicon": "https://static.xx.fbcdn.net/rsrc.php/yo/r/iRmz9lCMBD2.ico",
            "id": 4,
            "siteName": "Facebook",
            "siteUrl": "www.facebook.com",
            "sortIndex": 2,
          },
        ]
      }
      handleShowDialModal={[Function]}
    />
  </Connect(Component)>
</ThemeProvider>
`;
